packages/rulesets/src/oas/functions/oasDiscriminator.ts

```diff
@@ -0,0 +1,37 @@
+import type { IFunction, IFunctionResult } from '@stoplight/spectral-core';
+import { isObject } from './utils/isObject';
+
+export const oasDiscriminator: IFunction = (schema, _opts, { path }) => {
```

Comment:
Could probably use `createRulesetFunction` from `@stoplight/spectral-core` to simplify things a bit here, but it's OK the way it is.
Examples:
https://github.com/stoplightio/spectral/blob/develop/packages/functions/src/alphabetical.ts#L42
https://github.com/stoplightio/spectral/blob/e71cf134866e22665a2cb4d8635d1b4c98a5ed95/docs/guides/5-custom-functions.md#validating-options